spec/models/teacher_subject_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe TeacherSubject, type: :model do
+  context 'when validating presence' do
+    it { is_expected.to validate_presence_of(:rc_teacher_id) }
+    it { is_expected.to validate_presence_of(:subject_id) }
```

Comment:
No need to validate the foreign key. You already have the test for the associations